src/org/zaproxy/zap/extension/pscanrulesBeta/CSRFCountermeasures.java

```diff
@@ -148,14 +145,16 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 			
 			for (Element formElement : formElements) {
 				logger.debug("FORM ["+ formElement + "] has parent ["+ formElement.getParentElement()+"]");
+				StringBuilder sb = new StringBuilder();
```

Comment:
New changes looks good, but upon further analysis of the code we can do some more clean ups (if you don't mind):
1.  `sb` can be removed, we can just use `sbForm`;
2. `evidence` can be declared later, see further comments.

Reply:
All good, Happy to tackle more.